ledger/ledger-api-tests/suites/src/main/scala/com/daml/ledger/api/testtool/suites/v1_dev/ExplicitDisclosureIT.scala

```diff
@@ -576,39 +578,34 @@ final class ExplicitDisclosureIT extends LedgerTestSuite {
       } yield {
         assertGrpcError(
           errorDuplicateContractId,
-          LedgerApiErrors.CommandExecution.Interpreter.InvalidArgumentInterpretationError,
-          // TODO ED: Ensure contractId inlined in error message
-          Some("Found duplicated contract IDs in submitted disclosed contracts for template"),
+          LedgerApiErrors.CommandExecution.Preprocessing.PreprocessingFailed,
+          Some(
+            s"Preprocessor encountered a duplicate disclosed contract ID ContractId(${testContext.disclosedContract.contractId})"
+          ),
           checkDefiniteAnswerMetadata = true,
         )
-        assertGrpcError(
+
+        val expectedKeyHashString = {
+          val bytes = Bytes.fromByteString(
+            testContext.disclosedContract.metadata
+              .getOrElse(fail("metadata not present"))
+              .contractKeyHash
+          )
+          Hash.fromBytes(bytes).getOrElse(fail("Could not decode hash")).toHexString
+        }
+        assertGrpcErrorRegex(
           errorDuplicateKey,
           LedgerApiErrors.CommandExecution.Interpreter.InvalidArgumentInterpretationError,
-          // TODO ED: Ensure contract key hash inlined in error message
-          Some("Found duplicated contract keys in submitted disclosed contracts"),
+          Some(
+            Pattern.compile(
+              s"Found duplicated contract keys in submitted disclosed contracts .* $expectedKeyHashString"
+            )
+          ),
           checkDefiniteAnswerMetadata = true,
         )
       }
   })
 
-  // TODO ED: Deduplicate with CKLocalKeyVisibility
-  test(
-    "EDLocalKeyVisibility",
-    "A local contract can be fetched/looked-up -by-key when the readers are not the contracts' stakeholders",
-    allocate(SingleParty, SingleParty),
-    enabled = _.explicitDisclosure,
-  )(implicit ec => {
-    case Participants(Participant(ledger1, party1), Participant(ledger2, party2)) =>
-      import Test.LocalKeyVisibilityOperations
-      for {
-        ops <- ledger1.create(party1, LocalKeyVisibilityOperations(party1, party2))
-
-        _ <- synchronize(ledger1, ledger2)
-        _ <- ledger2.exercise(party2, ops.exerciseLocalLookup())
-        _ <- ledger2.exercise(party2, ops.exerciseLocalFetch())
-      } yield ()
-  })
```

Comment:
This test is now replaced by `ContractKeysIT:CKLocalLookupByKeyVisibility` and `ContractKeysIT:CKLocalFetchByKeyVisibility`